### ISSUE TITLE
Incorrectly flagged "String literal is unterminated" in f-string

### ISSUE THREAD

**Issue by fluggo** [microsoft/pyright#452]
Pyright gets confused with embedded strings and substrings in f-strings:

```python
use_extra_fields = True
field_list = f"[{{name,ipv4,tags}}{',createdDate,lastUpdatedDate,extra.{metadata}' if use_extra_fields else ''}]"

print(field_list)
```

Pyright complains that:

* `String literal is unterminated` for the string starting `',createdDate`...
* `'metadata' is not defined` even though `metadata` is part of the string and not part of an embedded expression

...but the code does in fact parse and run, and prints: 

```
[{name,ipv4,tags},createdDate,lastUpdatedDate,extra.{metadata}]
```

**VS Code extension or command-line**
This is the ms-pyright extension version 1.1.13 in VS Code.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by fluggo** [comment]
Another test sample for you, this time with an f-string inside an f-string:

```python
class A:
    def __repr__(self):
        return f'''{self.__class__.__name__} @{id(self) & 0xFFFFFF:x} {' '.join(f'{k}={v!r}' for k, v in self.__dict__.items())}'''
```

**Comment by erictraut** [comment]
Thanks for the additional test case. I already included a nested f-string example as a unit test, but this gives me additional confidence that I handled all the edge cases. :)

**Comment by erictraut** [comment]
This is now fixed in 1.1.14, which I just published.

**Comment by fluggo** [comment]
Verified, thanks much!
